components/load_from_hf_hub/src/main.py

```diff
@@ -44,12 +48,27 @@ def load(self) -> dd.DataFrame:
                 )
 
         # 3) Rename columns
+        logger.info("Renaming columns...")
         dask_df = dask_df.rename(columns=self.column_name_mapping)
 
         # 4) Optional: only return specific amount of rows
-        if self.n_rows_to_load:
-            dask_df = dask_df.head(self.n_rows_to_load)
-            dask_df = dd.from_pandas(dask_df, npartitions=1)
+        if self.n_rows_to_load is not None:
+            if self.dataset_length is None:
+                raise ValueError("""Make sure to also specify the length of the entire
+                                 dataset. This is required as otherwise only the first
+                                 partition can be loaded""")
+            logger.info(f"Loading approximately {self.n_rows_to_load} rows...")
+            partition_length = self.dataset_length // dask_df.npartitions
+            npartitions = self.n_rows_to_load // partition_length
+            dask_df = dask_df.head(self.n_rows_to_load, npartitions=npartitions)
+            dask_df = dd.from_pandas(dask_df, npartitions=npartitions)
+            # .reset_index(drop=True) # will reset it from 0 for every partition
+
+        # Set monotonically increasing index
+        logger.info("Setting the index...")
+        dask_df["id"] = 1
+        dask_df["id"] = dask_df.id.cumsum()
+        dask_df = dask_df.set_index("id", sort=True)
```

Comment:
This is based on https://stackoverflow.com/questions/47571715/dask-create-strictly-increasing-index